contracts/StakeLockContract.sol

```diff
@@ -28,74 +29,78 @@ contract StakeLockContract {
      */
     mapping(address => lockedToken) private locked;
     mapping(address => uint) private indices;
-    address[] private accountList;
+    address[] private addressList;
 
     event Locked(address indexed _of, uint _amount, uint256 _epoch);
     event Unlocked(address indexed account, uint index);
 
-    /**
-     * @dev Returns total tokens held by an address (locked + transferable)
-     * @param _of The address to query the total balance of
-     */
-    function lockedBalanceOf(address _of)
-        public
-        view
-        returns (uint256 amount)
-    {
-        amount = locked[_of]._amount;
-    }
-
     /**
      * @dev Locks a specified amount of tokens against an address
      *      starting at the specific epoch
-     * @param _epoch The epoch where the lock happens
      */
-    function lock(uint256 _epoch)
+    function lock()
         public
         payable
         returns (bool)
     {
         // If tokens are already locked, then functions extendLock or
         // increaseLockAmount should be used to make any changes
-        require(lockedBalanceOf(msg.sender) == 0, ALREADY_LOCKED);
+        require(balanceOf(msg.sender) == 0, ALREADY_LOCKED);
         require(msg.value != 0, AMOUNT_ZERO);
 
-        locked[msg.sender] = lockedToken(msg.value, block.number, currentEpoch(), 1);
+        // By default, the tokens can only be locked for one locking period.
+        locked[msg.sender] = lockedToken(msg.value, block.number, currentEpoch(), 1, addressList.push(msg.sender) - 1);
```

Comment:
what if the same address calls the lock function twice?

Reply:
it will throw ALREADY_LOCKED error